SimPEG/Directives.py

```diff
@@ -286,17 +287,127 @@ def endIter(self):
+        phi_m_smoomth_y = np.nan
+        phi_m_smoomth_z = np.nan
+
+        if self.reg.regmesh.dim == 2:
+            phi_m_smoomth_y = reg.objfcts[2](self.invProb.model) * self.reg.alpha_y
+        elif self.reg.regmesh.dim == 3:
+            phi_m_smoomth_y = self.reg.objfcts[2](self.invProb.model) * self.reg.alpha_y
+            phi_m_smoomth_z = self.reg.objfcts[3](self.invProb.model) * self.reg.alpha_z
+
+        f.write(' {0:3d} {1:1.4e} {2:1.4e} {3:1.4e} {4:1.4e} {5:1.4e} {6:1.4e}  {7:1.4e}  {8:1.4e}\n'.format(
+            self.opt.iter,
+            self.invProb.beta,
+            self.invProb.phi_d,
+            self.invProb.phi_m,
+            phi_m_small,
+            phi_m_smoomth_x,
+            phi_m_smoomth_y,
+            phi_m_smoomth_z,
+            self.opt.f))
         f.close()
 
+    def load_results(self):
+        results = np.loadtxt(self.fileName+str(".txt"), comments="#")
+        self.iterations = results[:, 0]
+        self.beta = results[:, 1]
+        self.phi_d = results[:, 2]
+        self.phi_m = results[:, 3]
+        self.phi_m_small = results[:, 4]
+        self.phi_m_smooth_x = results[:, 5]
+        self.phi_m_smooth_y = results[:, 6]
+        self.phi_m_smooth_z = results[:, 7]
+
+        if self.reg.regmesh.dim == 1:
+            self.phi_m_smooth = self.phi_m_smooth_x.copy()
+        elif self.reg.regmesh.dim == 2:
+            self.phi_m_smooth = self.phi_m_smooth_x + self.phi_m_smooth_y
+        elif self.reg.regmesh.dim == 3:
+            self.phi_m_smooth = (
+                self.phi_m_smooth_x + self.phi_m_smooth_y + self.phi_m_smooth_z
+                )
+
+        self.f = results[:, 7]
+
+        self.target_misfit = self.invProb.dmisfit.prob.survey.nD / 2.
+        self.i_target = None
+
+        if self.invProb.phi_d < self.target_misfit:
+            i_target = 0
+            while self.phi_d[i_target] > self.target_misfit:
+                i_target += 1
+            self.i_target = i_target
+
+    def plot_misfit_curves(self, fname=None, plot_small_smooth=False):
+        fig = plt.figure(figsize=(5, 2))
+        ax = plt.subplot(111)
+        ax_1 = ax.twinx()
+        ax.semilogy(self.iterations, self.phi_d, 'k-', lw=2)
+        ax_1.semilogy(self.iterations, self.phi_m, 'r', lw=2)
+        if plot_small_smooth:
+            ax_1.semilogy(self.iterations, self.phi_m_small, 'ro')
+            ax_1.semilogy(self.iterations, self.phi_m_smooth, 'rx')
+            ax_1.legend(
+                ("$\phi_m$", "small", "smooth"), bbox_to_anchor=(1.5, 1.)
+                )
+
+        ax.plot(np.r_[ax.get_xlim()[0], ax.get_xlim()[1]], np.ones(2)*self.target_misfit, 'k:')
+        ax.set_xlabel("Iteration")
+        ax.set_ylabel("$\phi_d$")
+        ax_1.set_ylabel("$\phi_m$", color='r')
+        for tl in ax_1.get_yticklabels():
+            tl.set_color('r')
+        plt.show()
+
+
+    def plot_tikhonov_curves(self, fname=None, dpi=200):
```

Comment:
this should eventually be a util, and maybe attached to this directive, but for now I think it is ok